packages/flame/lib/src/experimental/camera_component.dart

```diff
@@ -115,4 +118,51 @@ class CameraComponent extends Component {
   /// This variable helps prevent infinite recursion when a camera is set to
   /// look at the world that contains that camera.
   static int maxCamerasDepth = 4;
+
+  /// Makes the [viewfinder] follow the given [target].
+  ///
+  /// This method adds a [FollowBehavior] to the viewfinder. If there is another
+  /// [FollowBehavior] currently applied to the viewfinder, it will be removed
+  /// first.
+  ///
+  /// Parameters [maxSpeed], [horizontalOnly] and [verticalOnly] have the same
+  /// meaning as in the [FollowBehavior.new] constructor.
+  ///
+  /// If [snap] is true, then the viewfinder's starting position will be set to
+  /// the target's current location. If [snap] is false, then the viewfinder
+  /// will move from its current position to the target's position at the given
+  /// speed.
+  void follow(
```

Comment:
It would be good to have a `followVector2` too which can be used by the `BodyComponent` for example, since that one can't have a `PositionProvider` since the `body.position` shouldn't be changed.

Reply:
My thinking here is that if `target` is a `PositionProvider`, then it can really be tied to anything: to another component, to a particle system, to a mouse cursor, to a geometric center of a swarm of components, to an abstract calculation, etc.
The fact that `body.position` shouldn't change is really not an obstacle, since we can declare a no-op setter in this case.

Reply:
I created the `APositionProvider` class to make this process easier.

Reply:
That is quite a strange name. :sweat_smile: 
Is this class really necessary? We can just implement the `PositionProvider` on `BodyComponent` and do a no-op + assert in the setter just like you said.

Reply:
It's helpful if you want to create a new `PositionProvider` on the fly. Say,
```dart
camera.follow(APositionProvider(getValue: () => (player1.position + player2.position)/2));
```
or
```dart
camera.follow(APositionProvider(getValue: () => player.position + player.velocity*0.5));
```

Reply:
I think we can start doing basic implementations of the providers if we see a real need for it, I think it will confuse people more than it will help them currently (and it is still very simple to create a new class that implements the provider with the information that they would want if they know what they are doing).

Reply:
Why would it confuse people if they are not even required to use this class? It's just a handy helper. If somebody thinks this helper is not really helpful, then they can use whatever other solution they prefer.

Reply:
I think it is mostly the name that I don't like. Maybe the old school java naming convention would be better here, `PositionProviderImpl`?

Reply:
ok, renamed